dracut/99setup-root/initrd-setup-root

```diff
@@ -31,6 +31,34 @@ function walksysroot() {
 COREOS_BLANK_MACHINE_ID="42000000000000000000000000000042"
 MACHINE_ID_FILE="/sysroot/etc/machine-id"
 
+function selectiveosreset() {
+  local entry="/sysroot$1"
+  # Don't remove /sysroot itself
+  [ "${entry}" = "/sysroot" ] && return 0
+  [ "${entry}" = "/sysroot/" ] && return 0
+  # Don't remove the active /usr mount point
+  [ "${entry}" = "/sysroot/usr" ] && return 0
+  # Not really needed because find doesn't add a trailing slash but to be safe:
+  [ "${entry}" = "/sysroot/usr/" ] && return 0
+  if [ -d "${entry}" ]; then
+    # Try to delete dir, will fail if its contents are preserved
+    usrbin rmdir "${entry}" 2>/dev/null || true
+  else
+    rm "${entry}" || true # TODO: Report through flag file or fail the boot?
```

Comment:
I'd add `-f` so it won't fail on nonexistent files, but will still fail if there's a permission error or whatever. This is to say - we probably shouldn't be reporting anything or failing the boot when deleting nonexistent files.

```suggestion
    rm -f "${entry}" || true # TODO: Report through flag file or fail the boot?
```

Reply:
If `find` hands us wrong paths that don't exist something is off and I think it's nice for it to show up in `journalctl -u initrd-setup-root`. My current take on errors would be to not have a special report file for the motd warning but rely on system logs and never fail the boot.

Reply:
I've captured this in a comment now. I think failing the boot could lead to a boot loop and since failures in the initrd are also normally not easy to deal with for normal users (serial console plus the right timing for the debug session or even the GRUB menu if the initrd has a loop), I think we should always continue the boot and then it's regular debugging for the user by checking the journal.

Reply:
Sounds good. Then maybe drop the TODO and instead write that errors will be visible in the logs?

Reply:
Nevermind, you already did that.